contrib/sbase/Makefile

```diff
@@ -0,0 +1,312 @@
+TOPDIR = $(realpath ../..)
+
+.SUFFIXES:
+.SUFFIXES: .o .c
+
+HDR =\
+	arg.h\
+	compat.h\
+	crypt.h\
+	fs.h\
+	md5.h\
+	queue.h\
+	sha1.h\
+	sha224.h\
+	sha256.h\
+	sha384.h\
+	sha512.h\
+	sha512-224.h\
+	sha512-256.h\
+	text.h\
+	utf.h\
+	util.h
+
+LIBUTF = libutf.a
+LIBUTFSRC =\
+	libutf/fgetrune.c\
+	libutf/fputrune.c\
+	libutf/isalnumrune.c\
+	libutf/isalpharune.c\
+	libutf/isblankrune.c\
+	libutf/iscntrlrune.c\
+	libutf/isdigitrune.c\
+	libutf/isgraphrune.c\
+	libutf/isprintrune.c\
+	libutf/ispunctrune.c\
+	libutf/isspacerune.c\
+	libutf/istitlerune.c\
+	libutf/isxdigitrune.c\
+	libutf/lowerrune.c\
+	libutf/rune.c\
+	libutf/runetype.c\
+	libutf/upperrune.c\
+	libutf/utf.c\
+	libutf/utftorunestr.c
+
+LIBUTIL = libutil.a
+LIBUTILSRC =\
+	libutil/concat.c\
+	libutil/cp.c\
+	libutil/crypt.c\
+	libutil/ealloc.c\
+	libutil/enmasse.c\
+	libutil/eprintf.c\
+	libutil/eregcomp.c\
+	libutil/estrtod.c\
+	libutil/fnck.c\
+	libutil/fshut.c\
+	libutil/getlines.c\
+	libutil/human.c\
+	libutil/linecmp.c\
+	libutil/md5.c\
+	libutil/memmem.c\
+	libutil/mkdirp.c\
+	libutil/mode.c\
+	libutil/parseoffset.c\
+	libutil/putword.c\
+	libutil/reallocarray.c\
+	libutil/recurse.c\
+	libutil/rm.c\
+	libutil/sha1.c\
+	libutil/sha224.c\
+	libutil/sha256.c\
+	libutil/sha384.c\
+	libutil/sha512.c\
+	libutil/sha512-224.c\
+	libutil/sha512-256.c\
+	libutil/strcasestr.c\
+	libutil/strlcat.c\
+	libutil/strlcpy.c\
+	libutil/strsep.c\
+	libutil/strtonum.c\
+	libutil/unescape.c\
+	libutil/writeall.c
+
+BIN =\
+	basename\
+	cat\
+	chgrp\
+	chmod\
+	chown\
+	cmp\
+	comm\
+	cp\
+	cut\
+	dd\
+	dirname\
+	echo\
+	env\
+	expand\
+	expr\
+	false\
+	fold\
+	head\
+	link\
+	ln\
+	logname\
+	md5sum\
+	mkdir\
+	mkfifo\
+	mktemp\
+	mv\
+	nice\
+	nl\
+	nohup\
+	printenv\
+	pwd\
+	readlink\
+	renice\
+	rev\
+	rm\
+	rmdir\
+	seq\
+	setsid\
+	sha1sum\
+	sha224sum\
+	sha256sum\
+	sha384sum\
+	sha512sum\
+	sha512-224sum\
+	sha512-256sum\
+	sleep\
+	split\
+	sponge\
+	strings\
+	sync\
+	test\
+	tr\
+	true\
+	tsort\
+	tty\
+	unexpand\
+	uniq\
+	unlink\
+	uudecode\
+	uuencode\
+	wc\
+	which\
+	whoami\
+	xinstall\
+	yes\
+	cal\
+	cols\
+	cksum\
+	find\
+	grep\
+	join\
+	kill\
+	ls\
+	od\
+	paste\
+	pathchk\
+	printf\
+	sed\
+	sort\
+	tail\
+	tar\
+	tee
+
+BIN_DONT_WORK =\
```

Comment:
Please explain what are the show stoppers for each of binaries below. We can always provide userspace stubs for non-crucial system calls. Here's the list of currently missing system call implementations: https://mimiker.ii.uni.wroc.pl/source/xref/mimiker/lib/libc/sys/missing.S

Reply:
I recognize `date`, `touch` and `xargs` as frequently used. I'd really like to have them in working condition.

Reply:
Missing syscalls/functions for BIN_DONT_WORK tools are:

- `date` - missing `clock_settime` syscall
- `getconf` - `pathconf` syscall
- `time` - `getrusage` syscall and `_SC_CLK_TCK` argument for `sysconf`
- `touch` - `strptime` libc function. This also requires `tzalloc`, `tzfree`, `tzgetgmtoff`, `tzset` functions and `timezone`, `tzname` variables definitions
- xargs - requires `ARG_MAX` to be >=4096, however increasing the value to 65536 (default value from NetBSD) results in kernel panic (e.g while executing `# ls | xargs` in `/`).

Reply:
4. I managed to add the functions, but adding `tzname`, `timezone` variables is more complex - there are declarations in lib/libc/time/localtime.c and lib/libc/time/private.h, but they are disabled by some ifdefs (e.g https://github.com/cahirwpz/mimiker/blob/master/lib/libc/time/private.h#L529). Thus I need to figure out how to safely enable these declarations

Reply:
4. [strptime](https://pubs.opengroup.org/onlinepubs/9699919799/functions/strptime.html) is a part of POSIX standard. We need to add it sooner or later.

Reply:
4. I managed to add `tzname` and `timezone` variables - now `strptime` and thus `touch` work fine. However this required some considerable changes in `lib/libc/time/localtime.c` and `lib/libc/time/strptime.c` - https://github.com/cahirwpz/mimiker/pull/1361/commits/1bb5e4eabb2e761615fcc9af6f3fb04077cae1b7.